sdk/core/core/inc/az_str.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STR_H
+#define AZ_STR_H
+
+#include <az_assert.h>
+
+#include <az_static_assert.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <limits.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// A constant string.
+typedef struct {
+  // Points to the first character.
+  uint8_t const *begin;
+  // A numer of C characters (bytes) in the string.
+  size_t size;
+} az_const_str;
+
+typedef struct {
+  uint8_t* begin;
+  size_t size;
+} az_str;
+
+AZ_STATIC_ASSERT(CHAR_BIT == 8);
```

Comment:
you really don't need this. If your bytes aren't 8bit so many things will break that it's not worth checking for.

Reply:
Some platforms may have `CHAR_BIT` equal to 16. Some really really old platforms had `CHAR_BIT` is equal to 7. Currently, we are saying that this library can't work with such platforms. At least the UTF-8 string library and string literals.

Reply:
which platforms have it equal to 16?

Reply:
Some DSPs. I don't think we target them :-) but who knows how people will use the library.

Reply:
I also think we should pull this out. We can document that we expect all compilers to treat uint8_t as an unsigned 8-bit value.